test/cache.c

```diff
@@ -298,14 +298,11 @@ smoke_check_1(int express_test, unsigned paged)
     if (show_progress) /* 2 */
         fprintf(stdout, "%s() - %0d -- pass = %d\n", __func__, mile_stone++, (int)pass);
 
-    file_ptr = setup_cache((size_t)(4 * 1024 * 1024), (size_t)(2 * 1024 * 1024), paged);
+    file_ptr = setup_cache((size_t)(4 * 1024 * 1024), (size_t)(2 * 1024 * 1024), paged, &api_ctx);
 
     if (show_progress) /* 3 */
         fprintf(stdout, "%s() - %0d -- pass = %d\n", __func__, mile_stone++, (int)pass);
 
-    /* Push API context */
-    H5CX_push(&api_ctx);
```

Comment:
All of the push/pop calls in these tests appear to have been unnecessary, as API contexts are all pushed right after `setup_cache()` (which pushes its own API context) and popped right after `takedown_cache()` (which pops its own API context), at least before the initial version of this fix. Just removed all the extra push/pop calls and kept the single context pushed/popped by the setup/takedown functions.